jupyter_server/gateway/managers.py

```diff
@@ -293,10 +292,10 @@ def gateway_request(endpoint, **kwargs):
                        "Ensure gateway url is valid and the Gateway instance is running.".
                             format(GatewayClient.instance().url))
 
-    raise gen.Return(response)
+    return response
 
 
-class GatewayKernelManager(MappingKernelManager):
+class GatewayKernelManager(AsyncMappingKernelManager):
```

Comment:
Since the floor for jupyter_client 6.x has been unset, this must be restored to `MappingKernelManager`.

Reply:
We should first see if there's a reason to pin jupyter_client to 5.x. Usually it's good to pin as loose as possible, but if it makes us regress, there's no point.

Reply:
Let's not forget about this.  If left we won't see an issue during build tests since they'll be using an updated jupyter_client.  FWIW, 4479 leaves this as `MappingKernelManager` since the bowels are never used on the server.  It's merely a "foothold" from which kernel management is proxied to a Gateway server.

Reply:
This needs to be changed back to `MappingKernelManager`, otherwise we've disabled the use of a gateway on 3.5.  As mentioned in the previous comment, the superclass is not critical since `GatewayKernelManager` serves as a proxy.